src/components/MainRouter/MainRouter.test.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import MainRouter from "./MainRouter";
+import { createMemoryHistory } from "history";
+import { Router } from "react-router";
+import AuthProvider from "../../components/AuthProvider";
+
+const renderComponent = (initialAuthState: boolean) => (
+  initialPath: string
+) => {
+  const history = createMemoryHistory({ initialEntries: [initialPath] });
+  return {
+    history,
+    ...render(
+      <AuthProvider initialAuthState={initialAuthState}>
+        <Router history={history}>
+          <MainRouter />
+        </Router>
+      </AuthProvider>
+    ),
+  };
+};
+
+// TODO extend jest matchers instead
+const expectToBeOnPage = (
+  container: HTMLElement,
+  myPath: string,
+  expectedPage: string
+) => {
+  const expectedPath = `/${expectedPage}`;
+  expect(myPath).toEqual(expectedPath);
+  expect(container.querySelector(".ion-page")).toHaveAttribute(
+    "title",
+    expectedPage
+  );
+};
```

Comment:
If any of those helpers will be useful in the future, don't hesitate to put them in a helper file.

Reply:
Yeah good point, but I feel that at this point this function is not abstract enough to warrant its own file?